src/parser.ts

```diff
@@ -398,9 +398,7 @@ export class Parser extends DiagnosticEmitter {
               tn.range(declareStart, declareEnd), "declare"
             ); // recoverable
           }
-          if (!namespace) {
```

Comment:
This looks like an intentional check.

@dcodeIO Is there something behind this check, or can it be safely removed?

Reply:
I have taken a look for compiler part. It handles statements in namespace correctly.